### PR TITLE
Templates for deploying xenserver. Additional manual work required.

### PR DESCRIPTION
**Pull request by pujan14**
This patch provides required templates to provision xenserver with some additional work. Additional work is linked in readme file.

### CODE REVIEW

File: xenserver/xenserver-pxe.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+default xenserver
+label xenserver
+kernel mboot.c32
+append boot/XenServer-6.5-x86_64-xen.gz dom0_max_vcpus=16 dom0_mem=752M com1=115200,8n1 console=com1,vga --- boot/XenServer-6.5-x86_64-vmlinuz xencons=hvc console=hvc0 console=tty0 answerfile=<%= foreman_url('provision')%> install --- boot/XenServer-6.5-x86_64-initrd.img
```
Comment (ekohl) [inline xenserver/xenserver-pxe.erb:4]: I wonder if variables could be used here rather than hardcoding the version and arch.

**Comment by mmoll** [comment]
Thanks, I also did think about starting this, I hope to have a patch ready to add XenServer as an OS to Foreman core, so we can use the templates more flexible.

File: xenserver/xenserver-pxe.erb (hunk 1 above)
Comment (pujan14) [inline xenserver/xenserver-pxe.erb:4]: I am not sure if that's a good idea. General work flow for this setup to work is.
- Copy boot files from iso to tftp boot folder
- Create an http share and extract iso to that share (required for installation)
- Use these templates in foreman

I am not sure including this in foreman is a good idea or not. If it becomes part of standard foreman, there should be some sort of wizard or something that takes in iso and creates http share and copies required files into tight places.

**Comment by pujan14** [comment]
@mmoll are you working on patch to add XenServer as an OS to foreman core? If so I would like to help (not sure how, testing maybe?)

File: xenserver/xenserver-pxe.erb (hunk 1 above)
Comment (ekohl) [inline xenserver/xenserver-pxe.erb:4]: In Foreman you have a version, so if the guide describes to download a version and then enter the correct version, I think it's much more flexible.

Comment (pujan14) [inline xenserver/xenserver-pxe.erb:4]: Original files from ISO are named vmlinux.gz, vmlinuz and xen.gz. I renamed them on foreman host to right convention. User can copy them as it is and just use same name in template.

**Comment by mmoll** [comment]
testing will be required for sure :)

File: xenserver/xenserver-answerfile.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+  <installation mode="fresh" srtype="lvm">
+    <primary-disk gueststorage="<%= @host.params['gueststorage'] || 'yes' %>"><%= @host.params['primary-disk'] || 'sda' %></primary-disk>
+    <keymap>de</keymap>
+    <hostname><%= @host.shortname %></hostname>
+    <root-password type="hash"><%= root_pass %></root-password>
+    <source type="<%= @host.params['source-type'] %>"><%= @host.params['source-path'] %></source>
+    <admin-interface name="<%=@ host.primary_interface || 'eth0' %>" proto="static">
```
Comment (jonaz) [inline xenserver/xenserver-answerfile.erb:8]: synax error here? @ should be before host.

**Comment by mmoll** [comment]
continuation in #250

**Pull request closed without merging**